### PR TITLE
Customized docker script to launch instance of ES and AppSearch

### PR DESCRIPTION
**Pull request by andsel**
This PR override docker-compose used by CI to start one instance of Elasticsearch and one instance of AppSearch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +43,22 @@ bundle install
 bundle exec rspec
 ```
 
+#### CI in local machine
```
Comment (kares) [inline README.md:46]: :heart: thanks!

File: .ci/retrieve_credentials.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+function wait_for_as {
```
Comment (yaauie) [inline .ci/retrieve_credentials.sh]: Nit: `wait_for_appsearch`

Abbreviated names are great when you'll be typing them often, but long-form descriptive names convey _context_ to the reader.

Similar is true for environment variables used here and throughout the diff (`AS` -> `APPSEARCH`) and the name of this script (`retrieve_credentials` -> `retrieve_appsearch_credentials`)

Comment (andsel) [reply .ci/retrieve_credentials.sh]: Done

File: .ci/docker-compose.override.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#        - DISTRIBUTION=${DISTRIBUTION:-default}
+#        - DISTRIBUTION_SUFFIX=${DISTRIBUTION_SUFFIX}
+#    environment:
+#      - DISTRIBUTION=${DISTRIBUTION:-default}
+#      - DISTRIBUTION_SUFFIX=${DISTRIBUTION_SUFFIX}
+#      - INTEGRATION=${INTEGRATION:-false}
+#      - SECURE_INTEGRATION=${SECURE_INTEGRATION:-false}
+
+  elasticsearch:
+#    image: docker.elastic.co/elasticsearch/elasticsearch:$ELASTIC_STACK_VERSION
```
Comment (kares) [inline .ci/docker-compose.override.yml]: don't you want to use the env variable here?
noticed in the README guide that you do `export ELASTIC_STACK_VERSION=7.7.0`

Comment (andsel) [reply .ci/docker-compose.override.yml]: Your are right, was piece of config that I commented while tinkering with it

Comment (andsel) [reply .ci/docker-compose.override.yml]: Having the env var `ELASTIC_STACK_VERSION` read by config, when it has value like `7.x` the build fails. The question is, should the AppSearch and Elasticsearch version used aligned to this or it should be only Logstash that need to bring that version?

Comment (andsel) [reply .ci/docker-compose.override.yml]: for example in circleci config inside `app-search-php` they use a fixed version of the Elasticsearch and AppSearch (https://github.com/elastic/app-search-php/blob/master/.circleci/config.yml#L29) probably updating it on every release

File: .ci/docker-compose.override.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    ports:
+      - 9200:9200
+#    networks: ['stack']
+
+  appsearch:
+    build:
+      context: ../
+      dockerfile: .ci/Dockerfile.appsearch
+      args:
+        - ELASTIC_APPSEARCH_VERSION=$ELASTIC_STACK_VERSION
```
Comment (kares) [inline .ci/docker-compose.override.yml:48]: :stuck_out_tongue_winking_eye: anyhow I do like that we can potentially specify a different version

**Review by kares — APPROVED** [review]
LGTM CI :green_circle:

File: .ci/docker-compose.override.yml (hunk 4 above)
Comment (andsel) [reply .ci/docker-compose.override.yml:48]: Would like to have different AS end ES versions?

Comment (kares) [reply .ci/docker-compose.override.yml:48]: it's fine as is what I meant that we could potentially have a matrix where testing of ES+AS would not be the same version. which I do like (not sure it's relevant - maybe if we have some compatibility bugs in the plugin)

Comment (andsel) [reply .ci/docker-compose.override.yml:48]: I would separate that part in another PR

**Pull request merged**